### PR TITLE
FileSearch: Don't use RegExes, just do string comparisons.

### PR DESCRIPTION
**Pull request by waddlesplash**
Nothing used the RegEx feature of FileSearch, and GCC < 4.9 doesn't support C++11 RegEx properly, so get rid of it.

CC @comex as he wrote the feature & @lioncash for code review

### CODE REVIEW

File: Source/Core/Common/FileSearch.cpp [hunk 1]
```diff
@@ -37,20 +32,20 @@ static std::vector<std::string> FileSearchWithTest(const std::vector<std::string
...
-	for (const auto& str : globs)
-	{
-		if (regex_str.size() != 2)
-			regex_str += "|";
-		// convert glob to regex
-		regex_str += std::regex_replace(std::regex_replace(str, std::regex("\\."), "\\."), std::regex("\\*"), ".*");
-	}
-	regex_str += ")$";
-	std::regex regex(regex_str, std::regex_constants::icase);
+	bool accept_all = (exts.size() == 0) || (std::find(exts.begin(), exts.end(), "") != exts.end());
```
Comment (lioncash) [inline Source/Core/Common/FileSearch.cpp]: exts.empty()

File: Source/Core/DolphinWX/Config/InterfaceConfigPane.cpp [hunk 2]
```diff
@@ -162,7 +162,7 @@ void InterfaceConfigPane::LoadGUIValues()
 
 void InterfaceConfigPane::LoadThemes()
 {
-	auto sv = DoFileSearch({"*"}, {
```
Comment (comex) [inline Source/Core/DolphinWX/Config/InterfaceConfigPane.cpp]: I think this should be `{""}`

**Comment by comex** [comment]
why, why, why do we care about GCC < 4.9 :(

but I guess it isn't the end of the world if it wasn't being used anyway.

File: Source/Core/DolphinWX/Config/InterfaceConfigPane.cpp (hunk 2 above)
Comment (waddlesplash) [inline Source/Core/DolphinWX/Config/InterfaceConfigPane.cpp]: If you look at the new DoFileSearch, you'll see it treats an empty vector the same as a vector with just an empty string in it, so there's no need to do that.

**Comment by waddlesplash** [comment]
> why, why, why do we care about GCC < 4.9 :(

This should be a performance boost (though it may not be too noticeable) as well as supporting GCC < 4.9. And I want this to get Dolphin working on Haiku, as Haiku finally has EGL support, but is still on GCC 4.8 for the time being.

**Comment by waddlesplash** [comment]
Sorry, clarification - it'll only be a performance boost if the regex evaluator doesn't have a JIT. I'm not aware of any implementations that do have a JIT right now.

**Comment by JosJuice** [comment]
If you want to support GCC 4.8, aren't there more things you need to change? For instance, PR #2674 (and `emplace`s added after that PR was made)

**Comment by comex** [comment]
It's not a performance boost because the cost of doing a filename comparison is negligible - only the time it takes the OS to hand over the directory listing matters.

I guess 17 months isn't *that* long for Haiku to drag feet...

**Comment by waddlesplash** [comment]
> If you want to support GCC 4.8, aren't there more things you need to change? For instance, PR #2674 (and emplaces added after that PR was made)

Hmm, not sure. I know that SymbolDB at least compiled on the offline WIP I have with no errors or warnings, so I don't know...

**Comment by waddlesplash** [comment]
@JosJuice: It appears GCC 4.8 does have it: http://stackoverflow.com/questions/24389014/stdmap-emplace-gcc-4-8-2

File: Source/Core/DolphinWX/Config/InterfaceConfigPane.cpp (hunk 2 above)
Comment (comex) [inline Source/Core/DolphinWX/Config/InterfaceConfigPane.cpp]: Oh, don't do that.  Unnecessary special casing just makes code confusing.

**Comment by lioncash** [comment]
Even if it didn't it wouldn't be reverted just to support Haiku.

**Comment by waddlesplash** [comment]
I realize that.

**Comment by waddlesplash** [comment]
@comex fixed.

File: Source/Core/Common/FileSearch.cpp [hunk 3]
```diff
@@ -37,20 +32,17 @@ static std::vector<std::string> FileSearchWithTest(const std::vector<std::string
...
-	regex_str += ")$";
-	std::regex regex(regex_str, std::regex_constants::icase);
+	bool accept_all = std::find(exts.begin(), exts.end(), "") != exts.end();
 	return FileSearchWithTest(directories, recursive, [&](const File::FSTEntry& entry) {
-		return std::regex_match(entry.virtualName, regex);
+		if (accept_all)
+			return true;
+		std::string name = entry.virtualName;
+		std::transform(name.begin(), name.end(), name.begin(), ::tolower);
+		return std::any_of(exts.begin(), exts.end(), [&](std::string ext) {
```
Comment (lioncash) [inline Source/Core/Common/FileSearch.cpp]: const reference

**Comment by waddlesplash** [comment]
@lioncash fixed.

**Comment by phire** [comment]
No guarantee that gcc 4.8 will stay supported in the future, but LGTM.

**Pull request merged**